web_ui/src/pages/project-details/components/project-media/grid-media-item.component.tsx

```diff
@@ -56,59 +46,49 @@ export const GridMediaItem = ({
         <div
             id={`grid-media-item-${id}`}
             className={[classes.mediaItem, isSelected ? classes.mediaItemSelected : ''].join(' ')}
-            onPointerOver={handlePointerOver}
-            onPointerOut={handlePointerOut}
         >
-            {(isHovered || isSelected || selectedMediaItemAction !== undefined) && (
-                <Flex
-                    wrap
-                    ref={triggerRef}
-                    width={'100%'}
-                    position={'absolute'}
-                    justifyContent={'space-between'}
-                    UNSAFE_className={classes.mediaItemActionsWrapper}
-                >
-                    <View width='size-500' height='size-500' zIndex={4}>
-                        <Flex
-                            position='relative'
-                            justifyContent='center'
-                            alignItems='center'
-                            width='100%'
-                            height='100%'
-                        >
-                            <View
-                                width={'100%'}
-                                height={'100%'}
-                                position={'absolute'}
-                                borderRadius={'regular'}
-                                backgroundColor={'gray-50'}
-                            />
-                            <Checkbox
-                                aria-label={'Select media item'}
-                                isSelected={isSelected}
-                                onChange={toggleMediaSelection}
-                                UNSAFE_style={{ padding: 8 }}
-                            />
-                        </Flex>
-                    </View>
-
-                    <View
-                        backgroundColor={'gray-50'}
-                        borderRadius={'regular'}
-                        UNSAFE_style={{ boxSizing: 'border-box', display: 'flex', alignItems: 'center' }}
-                        paddingY={'size-150'}
-                        paddingX={'size-75'}
-                        height={'size-500'}
-                        zIndex={10}
-                    >
-                        <MediaItemActions
-                            mediaItem={mediaItem}
-                            onSelectedMediaItemActionChange={setSelectedMediaItemAction}
-                            selectedMediaItemAction={selectedMediaItemAction}
+            <Flex
```

Comment:
One thing to note here: I think here we were conditionally rendering based on `isHovered` and `isSelected` to improve the performance of the media grid.
One thing I've seen lately is that `Checkbox` can be quite slow to render which is more clear when you go to the media page and use many small thumbnails.

Reply:
I've done some testing and it looks like the lag I was thinking of is not noticeable on your environment. So I'm fine with sticking to this simpler solution.